blocks/library/image/block.js

```diff
@@ -132,60 +134,221 @@ class ImageBlock extends Component {
 	}
 
 	updateImageURL( url ) {
-		this.props.setAttributes( { url } );
+		this.props.setAttributes( { url, width: undefined, height: undefined } );
 	}
 
 	getAvailableSizes() {
 		return get( this.props.image, [ 'media_details', 'sizes' ], {} );
 	}
 
-	render() {
-		const { attributes, setAttributes, isSelected, className, settings, toggleSelection } = this.props;
-		const { url, alt, caption, align, id, href, width, height } = attributes;
+	renderBlockControls() {
```

Comment:
Minor: Just wanted to share my concerns about this pattern: (render*) functions. I think this probably indicates that we need to split this into smaller components. I think this pattern makes it hard to maintain the component without having an obvious pro.

Reply:
Also, this is something we'd want to be consistent across core blocks, because people will learn from them.

Reply:
Yeah, I also don't very much like this pattern. I opted for it instead of splitting the block into smaller components because there was a lot of codependency on local state. Looking at it now with fresh (post-Easter) eyes, though, it doesn't look like it would be so bad to pass props around.